storage/innobase/include/fil0crypt.h

```diff
@@ -327,6 +327,55 @@ struct fil_space_scrub_status_t {
   ulint current_scrub_max_page_number; /*!< current scrub max page no */
 };
 
+struct redo_log_key final {
+  uint version;
+  char key[ENCRYPTION_KEY_LEN];
+  ulint read_count;
+  ulint write_count;
+  bool present;
+
+  bool persisted() const noexcept { return version != 0; }
+};
+
+/** Handles the fetching/generation/storing/etc of keyring redo log keys.
+
+ This class is *NOT* thread safe, as thread safety is not required.
+ Data is only accessed/modified on the following points:
+ * When the redo space is created, at startup
+ * During redo log recovery, at startup
+ * When the server UUID is generated, at startup
+ * When the user requests a new key version, checked periodically in the
+   master thread
+
+ As these can't happen in parallel, no lock is used.
+ */
+class redo_log_keys final {
+ public:
+  MY_NODISCARD
+  redo_log_key *load_latest_key(bool try_generate = true);
```

Comment:
If possible, please remove default value and fix callers. Shouldn't be many. Also document, the parameter and the function.  Please explain when (or who) should use try_generate as true and who should use as false.

One more suggestion is to just use "generate" instead of "try_generate"

Reply:
Done.